pepper-apis/src/main/java/org/broadinstitute/ddp/route/GetActivityInstanceRoute.java

```diff
@@ -73,38 +78,37 @@ public Object handle(Request request, Response response) {
 
-            String preferredIsoLanguageCode = ddpAuth.getPreferredLanguage();
-            List<String> isoLangCodes = Arrays.asList(preferredIsoLanguageCode, DEFAULT_ISO_LANGUAGE_CODE);
-
             Optional<EnrollmentStatusType> enrollmentStatus = handle.attach(JdbiUserStudyEnrollment.class)
                     .getEnrollmentStatusByUserAndStudyGuids(userGuid, studyGuid);
-            // Return an activity instance translated to the first available language
-            LOG.info("Attempting to find a translation for the following languages: {}", isoLangCodes);
-            for (String isoLangCode : isoLangCodes) {
-                Optional<ActivityInstance> inst = actInstService.getTranslatedActivity(
-                        handle, userGuid, instanceDto.getActivityType(), instanceGuid, isoLangCode, style
-                );
-                if (inst.isPresent()) {
-                    LOG.info("Found a translation to the '{}' language code for the activity instance with GUID {}",
-                            isoLangCode, instanceGuid);
-                    ActivityInstance activityInstance = inst.get();
-                    // To-do: change this to just "if (enrollmentStatus.get() == EnrollmentStatusType.EXITED_BEFORE_ENROLLMENT)) {...}"
-                    // when every user registered in the system will become enrolled automatically
-                    // When it is implemented, the check for the enrollment status presence is not needed
-                    if (enrollmentStatus.isPresent() && enrollmentStatus.get().shouldMarkActivitiesReadOnly()) {
-                        activityInstance.makeReadonly();
-                    }
-                    // end To-do
-                    JdbiLanguageCode jdbiLanguageCode = handle.attach(JdbiLanguageCode.class);
-                    Long languageCodeId = jdbiLanguageCode.getLanguageCodeId(isoLangCode);
-                    return validateActivityInstance(handle, activityInstance, userGuid, languageCodeId);
-                } else {
-                    LOG.info("Failed to find a translation to the '{}' language code for the activity instance "
-                            + "with GUID {}, keep on searching", isoLangCode, instanceGuid);
-                }
+
+            Locale preferredUserLanguage = RouteUtil.resolvePreferredUserLanguage(
+                    handle, acceptLanguageHeader, ddpAuth.getPreferredLocale(), studyGuid
+            );
+            String isoLangCode = preferredUserLanguage.getLanguage();
+
+            LOG.info("Attempting to find a translation for the following language: {}", isoLangCode);
+            Optional<ActivityInstance> inst = actInstService.getTranslatedActivity(
+                    handle, userGuid, instanceDto.getActivityType(), instanceGuid, isoLangCode, style
+            );
+            if (!inst.isPresent()) {
+                LOG.info("Failed to find a translation to the '{}' language code for the activity instance "
```

Comment:
Is this a truly unexpected event and qualifies for a `log.error()`?

Reply:
Since this is just before a `throws`, let's just `throw`, since the logging will pick it up.

Reply:
Done.